homeassistant/components/github/sensor.py

```diff
@@ -114,6 +116,7 @@ def device_state_attributes(self):
             ATTR_NAME: self._name,
             ATTR_LATEST_COMMIT_MESSAGE: self._latest_commit_message,
             ATTR_LATEST_COMMIT_SHA: self._latest_commit_sha,
+            ATTR_LATEST_RELEASE_TAG: self._latest_release_tag,
```

Comment:
Only add the item to attributes if the value not `None`.

Reply:
Shouldn't we show an empty attribute field id value = None. I would have thought the attributes list should always be identical, independent of state/value?

Reply:
No, our standard is to only add the item if the value is not `None`.

Reply:
ok, how do I do that in this section?

Reply:
```py
attrs = {
    ...
}
if self._latest_release_tag is not None:
    attrs[ATTR_LATEST_RELEASE_TAG] = self._latest_release_tag
return attrs
```

Reply:
ok, though I understand the logic of this I am still confused where it should go. Do you want me to replace:

```
ATTR_LATEST_RELEASE_TAG: self._latest_release_tag,
```

with 

. so this full bit of code should go in line 119:
```
attrs = {
    ...
}
if self._latest_release_tag is not None:
    attrs[ATTR_LATEST_RELEASE_TAG] = self._latest_release_tag
return attrs
```

Reply:
The existing dictionary with attributes should be saved in `attrs`.

Reply:
ok, since I couldn't find any other config ike this in the home assistant GitHub components, I tried below, do you mean to have it like this:

```
@property
    def device_state_attributes(self):
        """Return the state attributes."""
        attrs = {
            ATTR_PATH: self._repository_path,
            ATTR_NAME: self._name,
            ATTR_LATEST_COMMIT_MESSAGE: self._latest_commit_message,
            ATTR_LATEST_COMMIT_SHA: self._latest_commit_sha,
            ATTR_LATEST_RELEASE_TAG: self._latest_release_tag,
            ATTR_LATEST_RELEASE_URL: self._latest_release_url,
            ATTR_LATEST_OPEN_ISSUE_URL: self._latest_open_issue_url,
            ATTR_OPEN_ISSUES: self._open_issue_count,
            ATTR_LATEST_OPEN_PULL_REQUEST_URL: self._latest_open_pr_url,
            ATTR_OPEN_PULL_REQUESTS: self._pull_request_count,
            ATTR_STARGAZERS: self._stargazers,
       }
      if self._latest_release_tag is not None:
          attrs[ATTR_LATEST_RELEASE_TAG] = self._latest_release_tag
      return attrs
```

Reply:
Yes, almost. Just remove `ATTR_LATEST_RELEASE_TAG` from `attrs` when creating it. That's the attribute we optionally will add.

Reply:
ok!